packages/amplify-provider-awscloudformation/lib/amplify-service-manager.js

```diff
@@ -0,0 +1,344 @@
+      } else if (appIdsInTheSameLocalProjectAndRegion.length > 1) {
+        context.print.info(`You project is associated with multiple Amplify Service Apps in the region ${region}`);
+        amplifyAppId = await SelectFromExistingAppId(context, appIdsInTheSameLocalProjectAndRegion);
+      }
+
+      if (!amplifyAppId) {
+        const createAppParams = {
+          name: projectConfig.projectName,
+        };
+        const createAppResponse = await amplifyClient.createApp(createAppParams).promise();
+        amplifyAppId = createAppResponse.app.appId;
+      }
+
+      const createEnvParams = {
+        appId: amplifyAppId,
+        environmentName: envName,
+        stackName: teamProviderInfo[envName][constants.ProviderName].StackName,
+        deploymentArtifacts: teamProviderInfo[envName][constants.ProviderName].DeploymentBucketName,
+      };
+
+      await amplifyClient.createBackendEnvironment(createEnvParams).promise();
+    }
+
+    teamProviderInfo[envName][constants.ProviderName][constants.AmplifyAppIdLabel] = amplifyAppId;
+    amplifyMeta.providers[constants.ProviderName][constants.AmplifyAppIdLabel] = amplifyAppId;
+
+    const amplifyMetaFilePath = context.amplify.pathManager.getAmplifyMetaFilePath();
+    let jsonString = JSON.stringify(amplifyMeta, null, 4);
+    fs.writeFileSync(amplifyMetaFilePath, jsonString, 'utf8');
+
+    const teamProviderInfoFilePath = context.amplify.pathManager.getProviderInfoFilePath();
+    jsonString = JSON.stringify(teamProviderInfo, null, 4);
+    fs.writeFileSync(teamProviderInfoFilePath, jsonString, 'utf8');
+  }
+}
+
+async function SelectFromExistingAppId(context, appIdsInTheSameLocalProjectAndRegion) {
+  let amplifyAppId;
+
+  const LEARNMORE = 'Learn More';
+  const NONE = 'None';
+
+  const options = appIdsInTheSameLocalProjectAndRegion.slice(0);
+  options.Push(NONE);
+  options.Push(LEARNMORE);
+
+  const answer = await inquirer.prompt({
+    type: 'list',
+    name: 'selection',
+    message: `Select the app id you want this env to be be associated with`,
+    choices: options,
+    default: options[0],
+  });
+
+  if (answer.selection === LEARNMORE) {
+    displayAppIdSelectionLearnMore(context);
+    amplifyAppId = await SelectFromExistingAppId(appIdsInTheSameLocalProjectAndRegion);
+  }
+
+  if (answer.selection !== NONE) {
+    amplifyAppId = answer.selection;
+  }
+
+  return amplifyAppId;
+}
+
+function displayAppIdSelectionLearnMore(context) {
+  // this should rarely happen
+  context.print.info('');
+  context.print.green("AWS Amplify Service stores information of your project's the backend environments in the cloud.");
+  context.print.green("It facilitates the team collaboration workflow of your project's development.");
+  context.print.green(
+    'You can select to associate this environment with an existing AWS Amplify Service App that other backend environements of the project are associated with.'
```

Comment:
Select an existing AWS Amplify Console app to associate this backend environment with the app.

Reply:
changed